### DISCUSSION TITLE
Making ImportAction Modal persistent

### DISCUSSION THREAD

**Discussion by KWangechi** [filamentphp/filament#12912] [Ideas]
Hey, I have a request, sometimes one can click outside the ImportAction modal by mistake and it closes by default. Is it possible to have it like a '$isPersistent' attribute in the main Importer Class and have it disappear when the 'Cancel' button is clicked?

Like have it as an option, with the current behavior being the default

**Comment by dissto** [comment]
[`->closeModalByClickingAway(false)`](https://filamentphp.com/docs/3.x/actions/modals#closing-the-modal-by-clicking-away) probably works on there too?

```php
ImportAction::make()
    ->closeModalByClickingAway(false)
```

**Comment by KWangechi** [comment]
Nice🔥

Thanks😊
